app/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/index.tsx

```diff
@@ -232,38 +218,51 @@ export const FormBranches = ({
                 defaultValues: defaultValues.referenceBranch || {},
               }}
             />
           )}
           {treatmentBranches &&
             showFirstTreatmentBranch &&
-            treatmentBranches.map(
-              (branch, idx) =>
-                branch && (
-                  <FormBranch
-                    {...{
-                      ...commonBranchProps,
-                      key: branch.key,
-                      fieldNamePrefix: `treatmentBranches[${idx}]`,
-                      //@ts-ignore react-hook-form types seem broken for nested fields
-                      errors: (errors?.treatmentBranches?.[idx] ||
-                        {}) as FormBranchProps["errors"],
-                      //@ts-ignore react-hook-form types seem broken for nested fields
-                      touched: (touched?.treatmentBranches?.[idx] ||
-                        {}) as FormBranchProps["touched"],
-                      branch,
-                      reviewErrors:
-                        (fieldMessages as SerializerMessages<SerializerSet[]>)
-                          .treatmentBranches?.[idx] || {},
-                      onRemove: handleRemoveBranch(idx),
-                      defaultValues:
-                        defaultValues.treatmentBranches?.[idx] || {},
-                    }}
-                  />
-                ),
-            )}
+            treatmentBranches.map((branch, idx) => {
+              const treatmentBranchFieldMessages = (
+                fieldMessages as SerializerMessages<SerializerSet[]>
+              ).treatmentBranches?.[idx];
+
+              return (
+                <FormBranch
+                  {...{
+                    ...commonBranchProps,
+                    key: branch.key,
+                    fieldNamePrefix: `treatmentBranches[${idx}]`,
+                    //@ts-ignore react-hook-form types seem broken for nested fields
+                    errors: (errors?.treatmentBranches?.[idx] ||
+                      {}) as FormBranchProps["errors"],
+                    //@ts-ignore react-hook-form types seem broken for nested fields
+                    touched: (touched?.treatmentBranches?.[idx] ||
+                      {}) as FormBranchProps["touched"],
+                    branch,
+                    reviewErrors:
+                      ({
+                        ...treatmentBranchFieldMessages,
+                        // Only show treatment branch feature config reviewErrors if the branch
+                        // has been saved or if other fields have errors, otherwise a required
+                        // feature config error can show on an optional branch
+                        ...((treatmentBranches?.[idx].slug ||
+                          treatmentBranchFieldMessages) && {
+                          featureConfig: fieldMessages.featureConfig,
+                        }),
+                      } as SerializerSet) || {},
```

Comment:
Forgot to say, I'm going to add a test for this too to check against the slug. If we ever change something in the BE I'd like to have this break since we rely on `slug` to check "if saved".